lib/shrine/plugins/logging.rb

```diff
@@ -57,6 +57,12 @@ def self.configure(uploader, opts = {})
       end
 
       module ClassMethods
+        # Assigns the subclass a copy of the logger.
+        def inherited(subclass)
+          super
+          subclass.opts[:logging_logger] = subclass.opts[:logging_logger].dup if subclass.opts[:logging_logger]
+        end
```

Comment:
After thinking about it more, I think it actually isn't a bad idea to share the `Logger` object between uploaders, in which case we could remove this code. I feel like that then we'd need to do the same in other plugins for consistency, and I don't see the need really. What do you think?

Reply:
I think a Logger instance should be shared in an app because I'd like to control logging level globally throughout the app. Then individual messages are controlled by using message level. Duplicating the logger in subclassing prevents the propagation of the global logger level to subclasses if done after subclassing. However I think it's ok to duplicate logger instance in multi-process/thread applications to maintain isolation between them.

What was your reasoning for duplicating logger in subclass? Are there any particular use case that calls for it?

As far as making other plugins consistent, I presume you mean the duplicating of their respective options or something else? I think it's nice to have consistency but that's a bigger change and we should review each plugin to see if it makes sense or not... something better done for next major release.

Reply:
> What was your reasoning for duplicating logger in subclass? Are there any particular use case that calls for it?

No, I thought it made sense, but it doesn't anymore.

> As far as making other plugins consistent, I presume you mean the duplicating of their respective options or something else? I think it's nice to have consistency but that's a bigger change and we should review each plugin to see if it makes sense or not... something better done for next major release.

Yeah, what I meant to say is that I'm in favour of *not* duplicating the `Logger` instance, but having subclasses use the same `Logger` instance. Because if we were to duplicate the `Logger` instance, that would not be consistent with any other potential plugin that's not duplicating it's option values. So I don't think it makes sense to duplicate the `Logger` instance at this point.

Reply:
Ok great so we're in agreement for not duplicating logger. I'll go ahead and make this change.